chainer/functions/connection/convolution_nd.py

```diff
@@ -0,0 +1,322 @@
+                gy_desc.value, gy.data.ptr, self.conv_desc.value,
+                zero.data, self.filter_desc.value, gW.data.ptr)
+            libcudnn.convolutionBackwardData_v2(
+                handle, one.data, self.filter_desc.value, W.data.ptr,
+                gy_desc.value, gy.data.ptr, self.conv_desc.value,
+                zero.data, x_desc.value, gx.data.ptr)
+
+        # Compute bias gradient if given and return gradients.
+        if b is None:
+            return gx, gW
+        else:
+            gb = cuda.cupy.empty_like(b)
+            libcudnn.convolutionBackwardBias(
+                handle, one.data, gy_desc.value, gy.data.ptr,
+                zero.data, self.bias_desc.value, gb.data.ptr)
+            return gx, gW, gb
+
+    def backward(self, inputs, grad_outputs):
+        x, W = inputs[:2]
+        b = inputs[2] if len(inputs) == 3 else None
+        gy = grad_outputs[0]    # (n, c_O, out_1, out_2, ..., out_N)
+
+        xp = cuda.get_array_module(*inputs)
+        if xp is numpy:
+            return self._backward_xp(x, W, b, gy, numpy)
+        elif not self._use_cudnn(x, W):
+            return self._backward_xp(x, W, b, gy, cuda.cupy)
+        else:
+            return self._backward_cudnn(x, W, b, gy)
+
+
+def convolution_nd(x, W, b=None, stride=1, pad=0, use_cudnn=True,
+                   cover_all=False):
+    """N-dimensional convolution function.
+
+    This is an implementation of N-dimensional convolution which is generalized
+    two-dimensional convolution in ConvNets. It takes three variables: the
+    input ``x``, the filter weight ``W`` and the bias vector ``b``.
+
+    Notation: here is a notation for dimensionalities.
+
+    - :math:`N` is the number of spatial dimensions.
+    - :math:`n` is the batch size.
+    - :math:`c_I` and :math:`c_O` are the number of the input and output
+      channels, respectively.
+    - :math:`d_1, d_2, ..., d_N` are the size of each axis of the input's
+      spatial dimensions, respectively.
+    - :math:`k_1, k_2, ..., k_N` are the size of each axis of the filters,
+      respectively.
+
+    Args:
+        x (~chainer.Variable): Input variable of shape
+            :math:`(n, c_I, d_1, d_2, ..., d_N)`.
+        W (~chainer.Variable): Weight variable of shape
+            :math:`(c_O, c_I, k_1, k_2, ..., k_N)`.
+        b (~chainer.Variable): One-dimensional bias variable with length
+            :math:`c_O` (optional).
+        stride (int or tuple of ints): Stride of filter applications
+            :math:`(s_1, s_2, ..., s_N)`. ``stride=s`` is equivalent to
+            ``(s, s, ..., s)``.
+        pad (int or tuple of ints): Spatial padding width for input arrays
+            :math:`(p_1, p_2, ..., p_N)`. ``pad=p`` is equivalent to
+            ``(p, p, ..., p)``.
+        use_cudnn (bool): If ``True``, then this function uses cuDNN if
+            available. cuDNN supports more than one-dimensional convolution.
```

Comment:
I am a little concern what `availabile` stands for. Some users may could be confused when they would encounter the situation in which `cuda.cudnn_enabled` is `True` but CuDNN is not used.